internal/server/polling/terraform.go

```diff
@@ -38,6 +62,119 @@ func (s *Server) getSource(ctx context.Context, tf *v1alpha2.Terraform) (*source
 	return obj, nil
 }
 
+func (s *Server) createTerraform(ctx context.Context, originalTF *infrav1.Terraform, originalSource *sourcev1b2.GitRepository, branch string, prID string, interval time.Duration) error {
+	tf := originalTF.DeepCopy()
+
+	tf.ResourceVersion = ""
+
+	tfName := s.createObjectName(tf.ObjectMeta.Name, branch, prID)
+
+	tf.SetName(tfName)
+
+	tf.SetLabels(s.createLabels(tf.Labels, branch, prID))
+
+	msg := fmt.Sprintf("Terraform object %s in the namespace %s", tfName, tf.ObjectMeta.Namespace)
+
+	source, err := s.createSource(ctx, originalSource, branch, prID, interval)
+	if err != nil {
+		return fmt.Errorf("unable to create Source for %s: %w", msg, err)
+	}
+
+	tf.Spec.SourceRef.Name = source.ObjectMeta.Name
+	tf.Spec.SourceRef.Namespace = source.ObjectMeta.Namespace
+
+	tf.Spec.PlanOnly = true
+	tf.Spec.StoreReadablePlan = "human"
+
+	tf.Spec.WriteOutputsToSecret.Name = s.createObjectName(tf.Spec.WriteOutputsToSecret.Name, branch, prID)
+
+	if err := s.clusterClient.Create(ctx, tf); err != nil {
+		return fmt.Errorf("unable to create %s: %w", msg, err)
+	}
+
+	s.log.Info(fmt.Sprintf("created %s", msg))
+
+	return nil
+}
+
+func (s *Server) createSource(ctx context.Context, originalSource *sourcev1b2.GitRepository, branch string, prID string, interval time.Duration) (*sourcev1b2.GitRepository, error) {
+	source := originalSource.DeepCopy()
+
+	source.ResourceVersion = ""
+
+	sourceName := s.createObjectName(source.ObjectMeta.Name, branch, prID)
+
+	source.SetName(sourceName)
+
+	source.Spec.Reference.Branch = branch
+
+	source.Spec.Interval = metav1.Duration{
+		Duration: interval,
+	}
+
+	source.SetLabels(s.createLabels(source.Labels, branch, prID))
+
+	msg := fmt.Sprintf("Source %s in the namespace %s", sourceName, source.ObjectMeta.Namespace)
+
+	if err := s.clusterClient.Create(ctx, source); err != nil {
+		return nil, fmt.Errorf("unable to create %s: %w", msg, err)
+	}
+
+	s.log.Info(fmt.Sprintf("created %s", msg))
+
+	return source, nil
+}
+
+func (s *Server) createObjectName(name string, branch string, prID string) string {
```

Comment:
[can be ignored]
It's fine here, but later we may want to move these helper functions out of the struct as the Informer can use it, maybe it will not use it at all, but I like taking extra notes.